addons/misra.py

```diff
@@ -740,8 +762,38 @@ def getForLoopCounterVariables(forToken):
                     vars_modified.add(tn.variable)
                 elif tn.previous and tn.previous.str in ('++', '--'):
                     vars_modified.add(tn.variable)
-        if cur_clause == 1 and tn.isAssignmentOp and tn.astOperand1.variable:
-            vars_initialized.add(tn.astOperand1.variable)
+        if cur_clause == 1 and tn.isAssignmentOp:
+            var_token = tn.astOperand1
+            while var_token and var_token.str == '.':
+                var_token = var_token.astOperand2
+            if var_token and var_token.variable:
+                vars_initialized.add(var_token.variable)
+        if cur_clause == 1 and tn.isName and tn.next.str == '(':
+            function_args_in_init = getArguments(tn.next);
+            function_args = {}
```

Comment:
not sure what is best but an alternative would be to first loop through the initfunction expressions..

        if cur_clause == 1 and tn.isName and tn.next.str == '(':
            function_args_in_init = getArguments(tn.next);
            for arg_nr in range(len(function_args_in_init))):
                init_arg = function_args_in_init[argnr]
                if init_arg is None or not init_arg.isUnaryOp('&'):
                    continue
                changed = False
                if tn.function is None:
                    changed = True # guess
                else:
                    ... determine if variable is changed ...
                if changed:
                    vars_initialized.add(get_variable(init_arg))